Bug #120430 fix: File type acceptance issue, upload & download (uploaded file), new param for file size

vaibhavsTekdi opened the pull request:
File type acceptance issue, upload & download (uploaded file), new param for file size

ankush-maherwal commented on administrator/models/field.php:
```diff
@@ -172,6 +172,7 @@ public function save_option($post)
 	{
 		$table = $this->getTable();
 		$data  = $post->get('jform', '', 'ARRAY');
+
```
remove extra line

ankush-maherwal commented on admin_language/en-GB/en-GB.com_tjfields.ini:
```diff
@@ -134,7 +134,7 @@ COM_TJFIELDS_FORM_LBL_EDITOR_HEIGHT="Height"
 COM_TJFIELDS_FORM_DESC_EDITOR_HEIGHT="Default Value"
 COM_TJFIELDS_FORM_LBL_FIELD_SIZE="Size"
-COM_TJFIELDS_FORM_DESC_FIELD_SIZE="Set size"
+COM_TJFIELDS_FORM_DESC_FIELD_SIZE="Set file size in MB (Numbers Only)"
```
Set maximum allowed file size (In MB) e.g 10, 20

ankush-maherwal commented on admin_language/en-GB/en-GB.com_tjfields.ini:
```diff
@@ -385,3 +385,12 @@ COM_TJFIELDS_PERMISSION_EDIT_FIELD_VALUE_DESC="Allows users in the group to edit
+COM_TJFIELDS_FORM_LBL_FIELD_COUNTERTEXT="Contertext"
+COM_TJFIELDS_FORM_DESC_FIELD_COUNTERTEXT="Format : Characters remaining {remaining} / {maxlength}"
+COM_TJFIELDS_FORM_LBL_FIELD_MAXLENGHT="Maxlength"
+COM_TJFIELDS_FORM_DESC_FIELD_MAXLENGHT="The maximum number of characters allowed in the text area"
```
Check the constants (Seems that file if not commited)

ankush-maherwal commented on administrator/models/fields/file.php:
```diff
@@ -128,6 +128,7 @@ protected function getInput()
 	{
 		$layoutData = $this->getLayoutData();
 		$html = '';
+		$html = $this->getRenderer($this->layout)->render($layoutData);
```
check the layout override with bootstrap 3 template  (File and Textarea counter)

ankush-maherwal commented on site/controller.php:
```diff
@@ -42,19 +51,24 @@ public function getMedia()
 		$query->select('*');
 		$query->from('#__tjfields_fields_value');
-		$query->quoteName('id') . ' = ' . $db->quote($file_id);
+		$query->where($db->quoteName('value') . " LIKE " . $db->Quote('%' . $fileName . '%'));
```
need an exact match, LIKE can cause uncertainties in the code

ankush-maherwal commented on site/controller.php:
```diff
@@ -65,23 +79,22 @@ public function getMedia()
 		{
 			$app->enqueueMessage(JText::_('JERROR_ALERTNOAUTHOR'), 'error');
-			$app->redirect($uri);
+			$app->redirect($this->returnURL);
```
do we need redirection in ajax call?

ankush-maherwal commented on site/helpers/tjfields.php:
```diff
@@ -205,20 +204,24 @@ public function saveFieldsValue($data)
 								{
 									$insert_obj_file->id = $if_edit_file_id;
-									$db->updateObject('#__tjfields_fields_value', $insert_obj_file, 'id');
+									$result = $db->updateObject('#__tjfields_fields_value', $insert_obj_file, 'id');
```
$result never used

ankush-maherwal commented on site/helpers/tjfields.php:
```diff
@@ -277,7 +280,7 @@ public function saveFieldsValue($data)
 							$db->setQuery($query);
 
-							$db->execute();
+							$result = $db->execute();
```
$result never used

ankush-maherwal commented on site/helpers/tjfields.php:
```diff
@@ -401,19 +404,50 @@ public function uploadFile($singleFile, $insert_obj_file, $file_field_data)
+				$filesizeInBytes = $this->formatSizeUnits($acceptSize);
+				if($fileSize > $filesizeInBytes)
+				{
+					$app->enqueueMessage('File size is greater than '. $acceptSize . 'MB' , 'warning');
```
use language constant

ankush-maherwal commented on site/helpers/tjfields.php:
```diff
@@ -401,19 +404,50 @@ public function uploadFile($singleFile, $insert_obj_file, $file_field_data)
 			$fileSize = $singleFile['size'];
 
-			if ($fileSize > 15728640)
+			if($acceptSize != "")
```
add pre-commit hooks check indentation

ankush-maherwal commented on site/helpers/tjfields.php:
```diff
@@ -401,19 +404,50 @@ public function uploadFile($singleFile, $insert_obj_file, $file_field_data)
 			}
+			else
+			{
+				if ($fileSize > 33554432)
```
file size should not be hard coded

ankush-maherwal commented on site/helpers/tjfields.php:
```diff
@@ -401,19 +404,50 @@ public function uploadFile($singleFile, $insert_obj_file, $file_field_data)
+			{
+				if ($fileSize > 33554432)
+				{
+					$app->enqueueMessage('File size is greater than 32MB', 'warning');
```
use language constant

ankush-maherwal commented on site/helpers/tjfields.php:
```diff
@@ -401,19 +404,50 @@ public function uploadFile($singleFile, $insert_obj_file, $file_field_data)
+			$acceptType = $fieldItems->params['accept'];
+			if (empty($acceptType))
+			{
+				$okMIMETypes    = 'pdf,PDF,doc,DOC,docx,DOCX,xls,XLS,xlsx,XLSX,jpeg,JPEG,png,PNG,jpg,JPG';
```
Why do we need hardcoded file types

ankush-maherwal commented on site/helpers/tjfields.php:
```diff
@@ -424,7 +458,7 @@ public function uploadFile($singleFile, $insert_obj_file, $file_field_data)
 			if (!in_array($fileMime, $validMIMEArray))
 			{
-				$app->enqueueMessage('This <strong>' . $title . '</strong> file type is not allowed for file ' . $filename . '.' . $extension, 'warning');
+				$app->enqueueMessage('Invalid file type', 'warning');
```
use language constant

ankush-maherwal commented on site/helpers/tjfields.php:
```diff
@@ -1601,24 +1633,42 @@ public function getMediaUrl($filePath, $extraUrlPrams = '')
-
+			// Here, fpht means file encoded path
+			$encodedPath = base64_encode($filePath);
+			$b_lk = 'index.php?option=com_tjfields&task=getMedia&fpht=';
```
use camel case and give a relevant name

vaibhavsTekdi commented:
Bug #120430 fix: File type acceptance issue, upload & download (uploaded file), new param for file size

Description : Removed the extra lines & spaces, changed the language constant, used the camel cases where it needed etc. in above commit.

will work on below points in next release :

1) Task #123196 : File Field : File size server validation using param or allowed upload max file size & post max size by server

2) Task #123199 File Field : File type server validation & need to set default file type accept criteria
